File: 2024/python/01/part2.py
v1 = []
v2 = []
with open("input.txt") as file:
    for line in file:
        numbers = line.strip().split()
        v1.append(int(numbers[0]))
        v2.append(int(numbers[-1]))
sum = 0
for i in v1:
    sum += i * v2.count(i)

print(sum)

File: 2024/perl/01/part2.pl
use strict;

open(FH, "input.txt") or die "couldn't open";
my @v1 = ();
my @v2 = ();
while (<FH>) {
    chomp;
    my ($a, $b) = split /\s+/;
    push(@v1,$a);
    push(@v2,$b);
}
my $sum = 0;

my %hm = ();
for (my $i = 0; $i < scalar(@v2); $i++) {
    $hm{$v2[$i]}++;
}
for (my $i = 0; $i < scalar(@v1); $i++) {
    $sum += $v1[$i] * $hm{$v1[$i]};
}
print $sum."\n";



